coderd/templates.go

```diff
@@ -139,17 +140,18 @@ func (api *API) postTemplateByOrganization(rw http.ResponseWriter, r *http.Reque
 		createTemplate                     codersdk.CreateTemplateRequest
 		organization                       = httpmw.OrganizationParam(r)
 		apiKey                             = httpmw.APIKey(r)
+		_, auditor                         = api.Auditor.Load(r.Context())
```

Comment:
Unfortunately, I don't think this construction helps much with the ergonomics of ensuring that handlers use a single feature implementation for their lifetime.  In 100% of the cases you've called `Load()` in product code, you've dropped the returned Context, and I can only assume that others will follow the same pattern.

I feel like the only thing that saves us here from calling `Load()` multiple times with the same context is the fact that we return 2 values so you can't inline it.

One thing that would help is to move it into middleware that sets the request context.  I would also like so see a middleware-based-getter (similar to `httpmw.APIKey(r)`) that panics if the middleware wasn't earlier in the handling chain.  If the endpoint handlers continue to call `Load()` it will be too easy to leave out the middleware without us noticing.

Reply:
This PR is blocking some Enterprise code from coming in, so I'd prefer to figure this out after we merge. The pattern before suffers from the same issue, so we aren't losing anything AFAIK.

I'll experiment with other patterns once this gets in (and I'm fine to simplify it right now too).